platform/mellanox/mlnx-platform-api/sonic_platform/component.py

```diff
@@ -469,6 +497,40 @@ def __install_firmware(self, image_path):
 
         return True
 
+    def auto_update_firmware(self, image_path, boot_action):
+        """
+        Handling of attempted automatic update for a SSD of a Mellanox switch.
+        Will first check the image_path to determine if a post-install reboot is required, 
+        then compares it against boot_action to determine whether to proceed with install.
+        """
+
+        # All devices support cold boot
+        supported_boot = ['cold']
+
+        # Verify image path exists
+        if not os.path.exists(image_path):
+            # Invalid image path
+            return FW_AUTO_ERR_IMAGE
+
+        # Check if post_install reboot is required
+        try:
+            if self.get_firmware_update_notification(image_path) is None:
+                # No power cycle required
+                supported_boot += ['warm', 'fast', 'none', 'any']
+        except RuntimeError:
+            # Unknown error from firmware probe
+            return FW_AUTO_ERR_UKNOWN
+
+        if boot_action in supported_boot:
+            if self.install_firmware(image_path):
+                # Successful update
+                return FW_AUTO_UPDATED
```

Comment:
Could it be more like below? If it doesn't need any complete action after installation, then it could be updated otherwise, it should return the installed status as we defined in hld.
```
 if self.get_firmware_update_notification(image_path) is None: 
    return FW_ATUO_UPDATED
else:
    return FW_AUTO_INSTALLED 
```

Reply:
No mellanox components need action at or after boot. We always do a full install and then some type of boot (warm / cold / power cycle) in all cases. 

Due to this we either install completely just prior to the reboot or we don't install at all, greatly simplifying this logic.

Reply:
I think if the reboot needs to be followed then the return should be "installed" not "updated".
"updated" means no additional action needed for the completion of the firmware update.

Reply:
Can you please point out to me where this is documented? I cannot seem to find it anywhere. I will make an update.

Reply:
@sujinmkang Is the updated vs installed flag used to do any logic? I'm a little confused at what the purpose is. For example, should there ever be an issue if I always pass installed as the return code so that the intended reboot is always followed?

Reply:
Please 2.2.2.4 update all commands at https://github.com/Azure/SONiC/blob/master/doc/fwutil/fwutil.md